tests/unit/s2n_self_talk_session_resumption_test.c

```diff
@@ -22,15 +22,105 @@
 #include <stdint.h>
 
 #include <s2n.h>
+#include <errno.h>
 
 #include "tls/s2n_connection.h"
 #include "tls/s2n_handshake.h"
 
-static const char SESSION_ID[] = "0123456789abcdef0123456789abcdef";
+
+#define MAX_KEY_LEN 32
+#define MAX_VAL_LEN 255
+
 static const char MSG[] = "Test";
 
+struct session_cache_entry {
+    uint8_t key[MAX_KEY_LEN];
+    uint8_t key_len;
+    uint8_t value[MAX_VAL_LEN];
+    uint8_t value_len;
+};
+
+struct session_cache_entry session_cache[256];
```

Comment:
You can probably use s2n_map for this (:

Reply:
checked map and we cannot use it here, because we cannot do a retrieve from the map until I complete it and that will not suite the cache use-case.